pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +190,58 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def pairwise(iterable: Iterable) -> Iterator[Tuple[Any, Any]]:
+    a, b = itertools.tee(iterable)
+    next(b, None)
+    return zip(a, b)
+
+
+def discriminator_encoder_modules(
+    in_channels: int = 3, inplace: bool = True
+) -> List[Tuple[str, nn.Sequential]]:
+    conv_block = partial(
+        ConvBlock, kernel_size=5, stride=2, padding="same", act="lrelu", inplace=inplace
+    )
+
+    channels = (in_channels, 128, 128, 256, 512, 512, 1024, 1024)
+    return [
+        (f"scale_{idx}", conv_block(in_channels, out_channels))
```

Comment:
Does the prefix `scale_` serve any purpose? If not, wouldn't be a simple `pystiche.SequentialModule` the better fit here?

Reply:
The prefix `scale_i` is used to select the scales `i` on which the `discriminator_loss` is calculated.

Reply:
Yeah, but what advantage does `f"scale_{idx}"` offer over `str(idx)` whih is the default of a sequential?

Reply:
It is the same name that was chosen in the the [reference_implementation](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/model.py#L134).

Reply:
If it does not serve any purpose, we should go with the simpler variant, since we aim to replicate the functionality and not each individual part.